nixos/modules/system/boot/luksroot.nix

```diff
@@ -594,6 +625,26 @@ in
             '';
           };
 
+          tryEmptyPassphrase = mkOption {
+            default = false;
+            type = types.bool;
+            description = ''
+              If keyFile fails then try an empty passphrase first before
+              prompting for password.
+            '';
+          };
+
+          keyFileTimeout = mkOption {
```

Comment:
Does this option need to be implemented in scripted stage 1? Or at least given documentation saying it's only effective in systemd stage 1?

Reply:
I'm not sure why this argument will need to be in scripted stage 1. We already check if systemd is enabled with the following [assertion](https://github.com/NixOS/nixpkgs/blob/dedb14591a1e54e79d7fb92f4bcf253ef7e0c1b9/nixos/modules/system/boot/luksroot.nix#L946). And this actually get's used [here](https://github.com/NixOS/nixpkgs/blob/dedb14591a1e54e79d7fb92f4bcf253ef7e0c1b9/nixos/modules/system/boot/luksroot.nix#L515)

Reply:
Actually, I think that assertion is wrong too. It says "When systemd stage 1 is enabled, all luks devices must have `dev.tryEmptyPassphrase` ***AND*** a non-null `dev.keyFileTimeout`," which would mean everyone always has to set it to true and have a timeout. So `dev.tryEmptyPassphrase && dev.keyFileTimeout != null` should be changed to `dev.tryEmptyPassphrase -> dev.keyFileTimeout != null` to say "All luks devices *that* have `dev.tryEmptyPassphrase` must have a non-null `dev.keyFileTimeout`."

Anyway, my point was that `keyFileTimeout` might as well be enabled for both systemd stage 1 and scripted stage 1, and it isn't implemented for scripted stage 1 here yet. But given that it's new functionality, it is not needed.

Reply:
Ah yeah good catch